src/cider/nrepl.clj

```diff
@@ -22,10 +22,24 @@
    [orchard.namespace]
    [orchard.java]))
 
-(when (and (< (-> *clojure-version* :major long) 2)
-           (< (-> *clojure-version* :minor long) 9))
-  (.println System/err "cider-nrepl cannot be run with older Clojure versions. Exiting.")
-  (System/exit 1))
+(def min-clojure-verion
+  "Having an enforced minimmum version can help users and maintainers alike diagnose issues more quickly,
+  avoiding problematic code paths in our middleware, and clients like cider.el."
+  {:major 1
+   :minor 9})
+
+(when (or (< (-> *clojure-version* :major long)
+             (-> min-clojure-verion :major long))
+
+          (and (= (-> *clojure-version* :major long)
+                  (-> min-clojure-verion :major long))
+               (< (-> *clojure-version* :minor long)
+                  (-> min-clojure-verion :minor long))))
+  (try
+    (.println System/err (format "cider-nrepl cannot be run with older Clojure versions (found: %s). Exiting."
```

Comment:
I still think it'd be good to mention here the required version as well, so it's clearer for the users what they need.

Reply:
Sure thing! https://github.com/clojure-emacs/cider-nrepl/commit/4497c166919d84fa6a6833f2baaf35cfe0687d14